### PR TITLE
f-self-exclusion@v1.0.0

### PR DESCRIPTION
**Pull request by m-axii**
Implemented Self-exclusion page component. Self-exclusion refers to the process where a customer opts to exclude themselves from an online alcohol delivery service.

Translations to currently in progress and will be implemented in a follow-up PR.

All commits are with --no-verify due to tests failing for f-checkout component, as agreed.

All tests for this component have been passed successfully.

### CODE REVIEW

File: package.json.orig [hunk 1]
```diff
@@ -0,0 +1,185 @@
+{
```
Comment (ashleynolan) [inline package.json.orig]: Looks like this is a git comparison diff file? Could you remove it so it doesn't get committed

File: packages/components/pages/f-self-exclusion/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+
+v0.1.0
+------------------------------
+*August 15, 2023*
+
+### Added
+- Add your change under the following headings: `Added`/`Changed`/`Deprecated`/`Removed`/`Fixed`/`Security`
+!!! Please, add your component to `.circleci/config.yml` under `save_cache_dist_directories` manually after the component is generated.
```
Comment (ashleynolan) [inline packages/components/pages/f-self-exclusion/CHANGELOG.md]:
```suggestion
*November 21, 2023*

### Added
- `f-self-exclusion` page created with base functionality
```

File: packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+@use "@justeat/fozzie/src/scss/fozzie" as f;
+
+.c-selfExclusion-card-component > div {
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    align-items: stretch;
+}
+
+.c-selfExclusion-title {
```
Comment (ashleynolan) [inline packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue]: I can't see a tag using this class – is it still relevant?

File: packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+            this.$emit('confirm-leave');
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+@use "@justeat/fozzie/src/scss/fozzie" as f;
+
+.c-selfExclusion-card-component > div {
```
Comment (ashleynolan) [inline packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue]: The base component class could just be simplified to `c-selfExclusion`.

File: packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+    flex-direction: column;
+    align-items: stretch;
+}
+
+.c-selfExclusion-title {
+    @include f.font-size(heading-m);
+}
+
+.c-selfExclusion-details,
+.c-selfExclusion-subtitle {
```
Comment (ashleynolan) [inline packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue]: Same as above – I can't see a tag using the `c-selfExclusion-subtitle` class – is it still relevant?

File: packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+    border: none;
+    padding: 0;
+}
+
+.c-selfExclusion-fieldset-date {
+    margin-left: f.spacing(f);
+    opacity: 0.5;
+}
+
+.c-selfExclusion-bottom-sheet-container {
```
Comment (ashleynolan) [inline packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue]: Just to fit with our component naming scheme:

```suggestion
.c-selfExclusion-bottomSheet {
```

File: packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+    @include f.media('<mid') {
+        position: relative;
+        bottom: 0;
+        left: 0;
+        right: 0;
+        margin-top: auto;
+    }
+}
+
+.c-buttons {
```
Comment (ashleynolan) [inline packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue]:
```suggestion
.c-selfExclusion-buttons {
```

File: packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+                <f-alert
+                    v-if="selectedState === 'temporaryExclusion' || selectedState === 'permanentExclusion'"
+                    type="warning"
+                    :heading="$t('alcoholSelfExclusionConfirmation.heading')"
+                >
+                    {{ selectedState === 'temporaryExclusion'
+                        ? $t('alcoholSelfExclusionConfirmation.text1Temporary')
+                        : $t('alcoholSelfExclusionConfirmation.text1Permanent') }}
+
+                    <p :class="$style['c-warning-text']">
```
Comment (ashleynolan) [inline packages/components/pages/f-self-exclusion/src/components/SelfExclusion.vue]:
```suggestion
                    <p :class="$style['c-selfExclusion-bottomSheet-warning']">
```

**Review by ashleynolan — COMMENTED** [review]
Few SCSS comments – otherwise looks good to me.

Also, could you remove the extra yarn.lock files that were committed? (there was a backup/base/local/remote file added with the PR)

**Review by ashleynolan — DISMISSED** [review]

File: packages/components/pages/f-self-exclusion/src/components/_tests/SelfExclusion.test.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    i18n,
+    $log
+} from './helpers/setup';
+
+const localVue = createLocalVue();
+
+localVue.use(VueI18n);
+localVue.use(Vuex);
+
+describe('SelfExclusion', () => {
```
Comment (jamieomaguire) [inline packages/components/pages/f-self-exclusion/src/components/_tests/SelfExclusion.test.js:16]: Based on what I can see in the template, it seems like there is some conditional rendering that could potentially be tested here as well as some user actions. Are there plans to add tests for these bits?

File: packages/components/pages/f-self-exclusion/src/services/providers/selfExclusion.api.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        this.#httpClient = httpClient;
+        this.#baseUrl = baseUrl;
+        this.#tenant = tenant;
+    }
+
+    async getExclusions (authToken) {
+        const headers = BuildHeaders(authToken);
+        const url = GET_EXCLUSIONS_URL.replace('{tenant}', this.#tenant);
+
+        const response = this.#httpClient.get(`${this.#baseUrl}/${url}`, headers);
```
Comment (jamieomaguire) [inline packages/components/pages/f-self-exclusion/src/services/providers/selfExclusion.api.js:28]: does the httpclient provide error handling? what happens if this call fails?

File: packages/components/pages/f-self-exclusion/src/services/providers/selfExclusion.api.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+        return response;
+    }
+
+    async updateAlcoholExclusion (authToken, body) {
+        const headers = BuildHeaders(authToken);
+        const url = PUT_ALCOHOL_EXCLUSION_URL.replace('{tenant}', this.#tenant);
+
+        const response = this.#httpClient.put(`${this.#baseUrl}/${url}`, body, headers);
```
Comment (jamieomaguire) [inline packages/components/pages/f-self-exclusion/src/services/providers/selfExclusion.api.js:37]: similar question here regarding error handling

File: packages/components/pages/f-self-exclusion/src/components/_tests/SelfExclusion.test.js (hunk 9 above)
Comment (m-axii) [reply packages/components/pages/f-self-exclusion/src/components/_tests/SelfExclusion.test.js:16]: No, we do not intend to add any tests for this component.

File: packages/components/pages/f-self-exclusion/src/services/providers/selfExclusion.api.js (hunk 10 above)
Comment (m-axii) [reply packages/components/pages/f-self-exclusion/src/services/providers/selfExclusion.api.js:28]: Will create an alert component to display the error.

File: packages/components/pages/f-self-exclusion/src/services/providers/selfExclusion.api.js (hunk 11 above)
Comment (m-axii) [reply packages/components/pages/f-self-exclusion/src/services/providers/selfExclusion.api.js:37]: We already have na alert displaying this error:
`            <f-alert
                v-else-if="isOpenAlertError"
                type="danger"
                :heading="$t('alcoholicItemsAlertError.heading')"
                is-dismissible>
                {{ $t('alcoholicItemsAlertError.text1') }}
            </f-alert>`

File: packages/components/pages/f-self-exclusion/src/components/_tests/SelfExclusion.test.js (hunk 9 above)
Comment (jamieomaguire) [reply packages/components/pages/f-self-exclusion/src/components/_tests/SelfExclusion.test.js:16]: This seems dangerous to me, but it's entirely up to you

**Review by jamieomaguire — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

**Pull request merged**